cmd/prometheus/main.go

```diff
@@ -1119,9 +1119,7 @@ func main() {
 						currentChecksum, err := config.GenerateChecksum(cfg.configFile)
 						if err != nil {
 							level.Error(logger).Log("msg", "Failed to generate checksum during configuration reload", "err", err)
-							continue
-						}
-						if currentChecksum == checksum {
+						} else if currentChecksum == checksum {
```

Comment:
`GenerateChecksum` can also return errors that have "nothing" to do with the config file being incorrect (finding/reading linked files issues etc)
Do we also want to reload on those?